IntegrationAdapters/Services/MedicineService.cs

```diff
@@ -30,6 +32,65 @@ public Medicine GetMedicineByName(String Name)
         public void AddMedicine()
         {
             medicineRepository.AddMedicineRepository();
-        }
+        }
+
+        public MedicineSpecification GetById(string id)
+        {
+            return medicineRepository.GetById(id);
+        }
+
+
+        private string GeneratePrescriptionString(PrescriptionDTO prescription)
+        {
+            string result = "";
+            result += prescription.PatientName + " " + prescription.PatientSurName + "\n\n";
+            result += prescription.Medicine + "\n";
+            result += prescription.Quantity + "\n";
+            result += prescription.PharmacyName + "\n";
+            result += prescription.Note + "\n";
+            return result;
+        }
+
+        public void GeneratePrescription(PrescriptionDTO prescription)
+        {
+            var fileName = "Prescription " + DateTime.Now.ToString("yyyy-MM-dd") + ".txt";
+            System.IO.File.WriteAllText(fileName, string.Empty);
+            TextWriter tw = new StreamWriter(fileName);
+            
+            tw.WriteLine(GeneratePrescriptionString(prescription));
+            tw.Close();
+        }
+
+        public void GenerateSpecificationFromPharmacy(string result, string MedicineName)

```

Comment:
Instead of result maybe name this parameter responseText so that it makes more sense?